### PR TITLE
Fix empty bucket creation error

### PR DESCRIPTION
**Pull request by iamyulong**
To allow creation of empty buckets in manifest

### CODE REVIEW

File: radix-engine/src/model/worktop.rs [hunk 1]
```diff
@@ -34,31 +34,29 @@ impl Worktop {
         &mut self,
         amount: Decimal,
         resource_address: ResourceAddress,
-    ) -> Result<Bucket, ResourceContainerError> {
+    ) -> Result<Option<Bucket>, ResourceContainerError> {
```
Comment (talekhinezh) [inline radix-engine/src/model/worktop.rs:37]: If returning `Option`, should just remove the `Result` part.

File: radix-engine/src/model/worktop.rs [hunk 2]
```diff
@@ -34,31 +34,29 @@ impl Worktop {
...
-    ) -> Result<Bucket, ResourceContainerError> {
-        self.take_non_fungibles(&BTreeSet::from([id.clone()]), resource_address)
-    }
-
     pub fn take_non_fungibles(
         &mut self,
         ids: &BTreeSet<NonFungibleId>,
         resource_address: ResourceAddress,
-    ) -> Result<Bucket, ResourceContainerError> {
+    ) -> Result<Option<Bucket>, ResourceContainerError> {
```
Comment (talekhinezh) [inline radix-engine/src/model/worktop.rs:52]: If returning Option, should just remove the Result part.

File: radix-engine/src/model/worktop.rs (hunk 1 above)
Comment (iamyulong) [reply radix-engine/src/model/worktop.rs:37]: I respectfully disagree though. `Option` means an optional return, while `Result` represents a fallible response. Not having the resource on worktop doesn't mean an error imo.

Comment (talekhinezh) [reply radix-engine/src/model/worktop.rs:37]: But there is no possible error in this method?

Comment (talekhinezh) [reply radix-engine/src/model/worktop.rs:37]: ie this method should be "infallible"

Comment (iamyulong) [reply radix-engine/src/model/worktop.rs:37]: There are many errors, e.g. invalid amount/granularity and insufficient balance.

Comment (iamyulong) [reply radix-engine/src/model/worktop.rs:37]: I think I missed one point though

Comment (iamyulong) [reply radix-engine/src/model/worktop.rs:37]: Should still check if the amount is zero before issuing a `None`.

Comment (talekhinezh) [reply radix-engine/src/model/worktop.rs:37]: Ahhh I see, didn't notice that `take_by_amount` is just mapped into the return value

**Review by talekhinezh — APPROVED** [review]

**Pull request merged**